Production/src/Selectors.cc

```diff
@@ -33,21 +68,25 @@ TauJetSelector::Result MuTau::Select(const edm::Event& event, const std::deque<T
                                      const std::vector<pat::Muon>& muons, const pat::MET& met,
                                      const reco::Vertex& primaryVertex,
                                      const pat::TriggerObjectStandAloneCollection& triggerObjects,
-                                     const edm::TriggerResults& triggerResults)
+                                     const edm::TriggerResults& triggerResults, const float rho)
 {
-    static const std::string filterName = "hltL3crIsoL1sMu22Or25L1f0L2f10QL3f27QL3trkIsoFiltered0p07";
+    static const std::string filterName = "hltL3crIsoL1sSingleMu22L1f0L2f10QL3f24QL3trkIsoFiltered0p07";
     static const std::set<int> decayModes = { 0, 1, 10, 11 };
 
+    // std::cout<< "In MuTau Selector" << std::endl;
+
     const pat::Muon *ref_muon = nullptr;
     for(const pat::Muon& muon : muons) {
-        if(!(muon.pt() > 28 && std::abs(muon.eta()) < 2.1 && muon.isMediumMuon() && PFIsolation(muon) < 0.15
+        if(!(muon.pt() > 25 && std::abs(muon.eta()) < 2.1 && muon.isMediumMuon() && PFRelIsolation(muon) < 0.15
                 && std::abs(muon.muonBestTrack()->dxy(primaryVertex.position())) < 0.2
                 && std::abs(muon.muonBestTrack()->dz(primaryVertex.position())) < 0.0045))
             continue;
-        if(!ref_muon || ref_muon->pt() < muon.pt())
+        if(!ref_muon || PFRelIsolation(*ref_muon) < PFRelIsolation(muon) || (PFRelIsolation(*ref_muon) == PFRelIsolation(muon) && ref_muon->pt() < muon.pt())){
```

Comment:
PFRelIsolation a < PFRelIsolation b means that a is more isolated than b. The condition should be inverted.